staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/virtualmachine.go

```diff
@@ -214,6 +214,29 @@ func (vm *VirtualMachine) IsActive(ctx context.Context) (bool, error) {
 	return false, nil
 }
 
+// Exists checks if VM exists and is not terminated
+func (vm *VirtualMachine) Exists(ctx context.Context) (bool, error) {
+	vmMoList, err := vm.Datacenter.GetVMMoList(ctx, []*VirtualMachine{vm}, []string{"summary.runtime.powerState"})
+	if err != nil {
```

Comment:
Do you anticipate not found fault here? if yes then we need to interpret it as VM not exist and return true.

Reply:
Hi @divyenpatel , it anticipated not found fault so it will return false as not Exist

Reply:
I created a custom binary and tested it out inside k8s environment. If VM is deleted, then node will be deleted as well

Reply:
```
I1111 06:26:45.027867       7 node_lifecycle_controller.go:773] Controller observed a new Node: "d5924270-2588-44f7-a44f-59063b5646f0"
I1111 06:26:45.027910       7 controller_utils.go:172] Recording Registered Node d5924270-2588-44f7-a44f-59063b5646f0 in Controller event message for node d5924270-2588-44f7-a44f-59063b5646f0
W1111 06:26:45.028007       7 node_lifecycle_controller.go:1044] Missing timestamp for Node d5924270-2588-44f7-a44f-59063b5646f0. Assuming now as a timestamp.
I1111 06:26:45.028023       7 node_lifecycle_controller.go:872] Node d5924270-2588-44f7-a44f-59063b5646f0 is NotReady as of 2020-11-11 06:26:45.02801646 +0000 UTC m=+470.966198179. Adding it to the Taint queue.
I1111 06:26:45.028342       7 event.go:291] "Event occurred" [object d5924270-2588-44f7-a44f-59063b5646f0 kind Node apiVersion v1 type Normal reason RegisteredNode message Node d5924270-2588-44f7-a44f-59063b5646f0 event: Registered Node d5924270-2588-44f7-a44f-59063b5646f0 in Controller]="(MISSING)"

```

```
I1111 06:34:40.072935       7 controller_utils.go:172] Recording Removing Node d5924270-2588-44f7-a44f-59063b5646f0 from Controller event message for node d5924270-2588-44f7-a44f-59063b5646f0
I1111 06:34:40.077904       7 event.go:291] "Event occurred" [object d5924270-2588-44f7-a44f-59063b5646f0 kind Node apiVersion v1 type Normal reason RemovingNode message Node d5924270-2588-44f7-a44f-59063b5646f0 event: Removing Node d5924270-2588-44f7-a44f-59063b5646f0 from Controller]="(MISSING)"
```

and volume attached to it are removed in the end
```
W1111 06:41:35.199254       7 reconciler.go:206] attacherDetacher.DetachVolume started for volume "pvc-e560795e-5736-4f0d-8ca3-48c1443d5909" (UniqueName: "kubernetes.io/vsphere-volume/[vsanDatastore] 64ac5c5e-1447-dc56-80b1-e4434bbb68dc/test2-dynamic-pvc-e560795e-5736-4f0d-8ca3-48c1443d5909.vmdk") on node "d5924270-2588-44f7-a44f-59063b5646f0" This volume is not safe to detach, but maxWaitForUnmountDuration 6m0s expired, force detaching
```